auto3dseg/algorithm_templates/segresnet/scripts/algo.py

```diff
@@ -34,292 +65,194 @@ def fill_template_config(self, data_stats_file, output_path, **kwargs):
-                'hyper_parameters.yaml': hyper_parameters,
-                'network.yaml': network,
-                'transforms_train.yaml': transforms_train,
-                'transforms_validate.yaml': transforms_validate,
-                'transforms_infer.yaml': transforms_infer
-                }
+                config["crop_mode"] = "rand"
+
+            ###########################################
+            # update network config
+            blocks_down = None
+            if levels >= 5:  # default
+                blocks_down = [1, 2, 2, 4, 4]
+            elif levels == 4:
+                blocks_down = [1, 2, 2, 4]
+            elif levels == 3:
+                blocks_down = [1, 3, 4]
+            elif levels == 2:
+                blocks_down = [2, 6]
+            elif levels == 1:
+                blocks_down = [8]
+
+            if blocks_down is not None:
+                config["network#blocks_down"] = blocks_down
+
+            config.update(input_config)  # override if any additional inputs provided
+
+            fill_records = {"hyper_parameters.yaml": config}
         else:
             fill_records = self.fill_records
 
         for yaml_file, yaml_contents in fill_records.items():
-            file_path = os.path.join(output_path, 'configs', yaml_file)
+            file_path = os.path.join(output_path, "configs", yaml_file)
 
             parser = ConfigParser(globals=False)
             parser.read_config(file_path)
             for k, v in yaml_contents.items():
                 if k in kwargs:
                     parser[k] = kwargs.pop(k)
                 else:
-                    parser[k] = deepcopy(v)  # some values are dicts
-                yaml_contents[k] = deepcopy(parser[k])
+                    parser[k] = copy.deepcopy(v)  # some values are dicts
+                yaml_contents[k] = copy.deepcopy(parser[k])
 
             for k, v in kwargs.items():  # override new params not in fill_records
-                if (parser.get(k, None) is not None):
-                    parser[k] = deepcopy(v)
+                if parser.get(k, None) is not None:
+                    parser[k] = copy.deepcopy(v)
                     yaml_contents.update({k: parser[k]})
 
-            ConfigParser.export_config_file(parser.get(), file_path, fmt="yaml", default_flow_style=None)
+            ConfigParser.export_config_file(
+                parser.get(), file_path, fmt="yaml", default_flow_style=None, sort_keys=False
+            )
 
         return fill_records
 
+    def export_to_disk(self, output_path: str, algo_name: str, **kwargs):
+        super().export_to_disk(output_path=output_path, algo_name=algo_name, **kwargs)
 
-if __name__ == "__main__":
-    from monai.utils import optional_import
+        output_path =os.path.join(output_path, algo_name)
+        config = ConfigParser.load_config_file(os.path.join(output_path, "configs/hyper_parameters.yaml"))
+
+        for c in config.get('custom_data_transforms',[]):
```

Comment:
The `custom_data_transforms` will be saved in the the `configs/hyper_parameters.yaml` of the generated algorithm folder (e.g. `segresent_0`). It means it will be used when users do the command-line-based trainings
`python segresnet_0/train.py` or `torchrun segresnet_0/train.py`

But I am worried that the following way will not load the `custom_data_transforms`
```
one_algo = algo_from_pickle('segresnet.pkl')
one_algo.train()
```
Because the process uses the templates and the `fill_records` for instantiation.

Reply:
thanks, I'm not exactly sure what you mean, it seems to work fine for me.  if you found it not working, please suggest a change.   But overall, we do have a tentative plan to remove pickling strategy in the future.

Reply:
I'm not sure how "custom_data_transforms" will look like. Something below could be a fix in the end of the reloaded `export_to_disk` functions.

```python
# to ensure the exported pickle file will include information about the custom_data_transform
self.fill_records['hyper_parameters.yaml']['custom_data_transforms'] = ...
```

Reply:
@myron Thanks for providing an example on Slack. This is resolved.